### PR TITLE
[SYCLomatic][CMake] Emit warning msg when CUDA-specific command is not supported during CMake script migration

### PR DESCRIPTION
**Pull request by tomflinda**
Signed-off-by: chenwei.sun <chenwei.sun@intel.com>

### CODE REVIEW

File: clang/lib/DPCT/MigrateCmakeScript.cpp [hunk 1]
```diff
@@ -614,14 +620,27 @@ static std::string convertCmakeCommandsToLower(const std::string &InputString) {
...
+      auto Iter = cmake_commands.find(Str);
+      if (Iter != cmake_commands.end()) {
         for (int Idx = Begin; Idx < End; Idx++) {
           Line[Idx] = Str[Idx - Begin];
         }
+        if (!Iter->second) {
+          std::string Prefix =
+              FileName + ":" + std::to_string(Count) + ":warning:";
+          std::string WaringContent =
+              "DPCT2000: migration of syntax \"" + Str +
```
Comment (zhiweij1) [inline clang/lib/DPCT/MigrateCmakeScript.cpp]: Is it better to move the message to:
```
//===--------------- DiagnosticsBuildScript.inc ---------------------------===//
//
// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
// See https://llvm.org/LICENSE.txt for license information.
// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
//
//===----------------------------------------------------------------------===//

// FORMAT: {NAME, ID, MESSAGE}
DEF_COMMENT(GEN_MAKEFILE_LIB, 2001, "You can link with more library by add them here.")
DEF_COMMENT(GEN_MAKEFILE_OPT, 2002, "You can change different compiler optimization options here.")
```
?

Comment (tomflinda) [reply clang/lib/DPCT/MigrateCmakeScript.cpp]: good advice, will be updated it..

File: clang/lib/DPCT/DiagnosticsCMakeScriptMigration.inc [hunk 2]
```diff
@@ -0,0 +1,10 @@
+//===--------------- DiagnosticsCMakeScriptMigration.inc ------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// FORMAT: {NAME, ID, MESSAGE}
+DEF_COMMENT(CMAKE_NOT_SUPPORT_WARNING, 3000, "migration of syntax \"{0}\" is not supported, you may need to adjust the code.")
```
Comment (zhiweij1) [inline clang/lib/DPCT/DiagnosticsCMakeScriptMigration.inc]:
```suggestion
DEF_COMMENT(CMAKE_NOT_SUPPORT_WARNING, 3000, "Migration of syntax \"{0}\" is not supported; you may need to adjust the code.")
```

**Review by zhiweij1 — APPROVED** [review]

File: clang/lib/DPCT/MigrateCmakeScript.cpp [hunk 3]
```diff
@@ -102,16 +103,19 @@ const std::unordered_map<std::string /*command*/, bool /*need lower*/>
...
         {"cuda_compile_ptx", 1},
         {"cuda_compile_fatbin", 1},
         {"cuda_compile_cubin", 1},
-        {"cuda_compute_separable_compilation_object_file_name", 1},
+        {"cuda_compute_separable_compilation_object_file_name", 0},
         {"cuda_include_directories", 1},
-        {"cuda_link_separable_compilation_objects", 1},
-        {"cuda_select_nvcc_arch_flags", 1},
-        {"cuda_wrap_srcs", 1},
+        {"cuda_link_separable_compilation_objects", 0},
```
Comment (zhimingwang36) [inline clang/lib/DPCT/MigrateCmakeScript.cpp]: Please isolate the map into a inc files, which we be edited and maintained easier, just like the API files.

Comment (tomflinda) [reply clang/lib/DPCT/MigrateCmakeScript.cpp]: Okay, I will implement it in the updated PR.

File: clang/lib/DPCT/CMakeCommands.inc [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+*           see if it has be migrated Commently.
+*
+*  MigrationDesc: Comment[: Comment] | Unsupported: Comment
+*    Comment:   this Command has been migrated.
+*    Unsupported:  this Command can not be migrated currently.
+*                  Must add Comment to explain why this API is Unsupported.
+*    Comment:      extra comment on the migration.
+*
+****************************************************************************/
+ENTRY_TYPE(cmake_minimum_required, true, "comment")
```
Comment (zhimingwang36) [inline clang/lib/DPCT/CMakeCommands.inc]: should we add one more field to indicate, it is CUDA-specific?   
if it is CUDA-specific and it's false on support, then it emits a warning message.

Comment (tomflinda) [reply clang/lib/DPCT/CMakeCommands.inc]: Good advice; it makes the inc file more friendly to read; I will add them.

File: clang/lib/DPCT/CMakeCommands.inc [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+ENTRY_TYPE(remove, true, "comment", false)
+ENTRY_TYPE(subdir_depends, true, "comment", false)
+ENTRY_TYPE(subdirs, true, "comment", false)
+ENTRY_TYPE(use_mangled_mesa, true, "comment", false)
+ENTRY_TYPE(utility_source, true, "comment", false)
+ENTRY_TYPE(variable_requires, true, "comment", false)
+ENTRY_TYPE(write_file, true, "comment", false)
+ENTRY_TYPE(cuda_add_cufft_to_target, true, "comment", false)
+ENTRY_TYPE(cuda_add_cublas_to_target, true, "comment", false)
+ENTRY_TYPE(cuda_add_executable, true, "comment", false)
```
Comment (zhimingwang36) [inline clang/lib/DPCT/CMakeCommands.inc]:
```suggestion
ENTRY_TYPE(cuda_add_executable, true, "comment", true)
```

Comment (tomflinda) [reply clang/lib/DPCT/CMakeCommands.inc]: Updated

**Review by ShengchenJ — APPROVED** [review]

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**